core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -485,13 +490,11 @@ class Partition(val topicPartition: TopicPartition,
   def makeLeader(partitionState: LeaderAndIsrPartitionState,
```

Comment:
I don't think AlterIsr changes anything since we're now just sending the async ISR update where we were previously directly updating ZK. 

Looking at the usages, `updateFollowerFetchState` is only called following a read (`Partition#readRecords`). These reads only happen on fetch requests and from the alter log dirs fetcher. I'm not sure about the alter log dirs flow, but as long as it sends the leader epoch, it should be safe.